internal/file/file.go

```diff
@@ -0,0 +1,101 @@
+package file
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/spf13/afero"
```

Comment:
Is there any reason to use this over `os.DirFS` and the `fs.FS` interface? I don't think we'll need to use network filesystems like the ones supported by afero.

Reply:
Removed afero, added https://pkg.go.dev/github.com/liamg/memoryfs for the tests